### PR TITLE
Handles exception thrown when the theme has not been initialised yet

### LINKED ISSUE #15300: NullPointerException: Attempt to invoke virtual method 'org.wordpress.android.fluxc.model.EditorThemeSupport org.wordpr...
- **Sentry Urls**: (events/users)
  - https://sentry.io/share/issue/67078fb4bbe14c8c8380f4114a2be02c/ (26/1)
  - https://sentry.io/share/issue/d38dd0d3aa4743619ea43a1655c0cd90/ (21/3)
  - https://sentry.io/share/issue/d1e705d510174c5fbba40142b05cf735/ (5/1)
  - https://sentry.io/share/issue/ca3efba9846147308537b7df17ed8d59/ (4/1)
  - https://sentry.io/share/issue/9622c255077b442d9e11c5f787c25a7d/ (2/1)
- **User Count**: 1 + 3 + 1 + 1 + 1 = 7
- **Count**: 26 + 21 + 5 + 5 + 2 = 59
- **First Release**: 18.2-rc-1
- **First Seen**: 2021-09-06T17:53:57.558000Z
- **Last Seen**: 2021-09-08T09:55:45Z
- **24 Hours**: 15 _(value extracted only from the first issue with 26/1)_
- **30 Days**: 26 _(value extracted only from the first issue with 26/1)_

```
NullPointerException: Attempt to invoke virtual method 'org.wordpress.android.fluxc.model.EditorThemeSupport org.wordpress.android.fluxc.model.EditorTheme.getThemeSupport()' on a null object reference
     at org.wordpress.android.ui.posts.EditPostActivity.onEditorFinalTouchesBeforeShowing(EditPostActivity.java:3305)
     at org.wordpress.android.ui.posts.EditPostActivity.onEditorFragmentInitialized(EditPostActivity.java:3283)
     at org.wordpress.android.editor.AztecEditorFragment.onCreateView(AztecEditorFragment.java:382)
...
(56 additional frame(s) were not displayed)
```

There seem to be multiple Sentry issues all with the similar exception for this one. Might be worth considering merging them (after double-checking they are indeed the same)

### PR DESCRIPTION
**Pull request by antonis**
Fixes #15300

**Based on** https://github.com/wordpress-mobile/WordPress-Android/pull/15308 and targeting `release/18.2`

## Description
This PR fixes a `NullPointerException` introduced in https://github.com/wordpress-mobile/WordPress-Android/pull/15134

To test:
I was not able to reproduce the conditions that lead to a crash. Thus I only validated the fix by triggering a `null` value for the `editorTheme` at debug time.

A few edge cases I tested include:
* Starting the editor quickly after a clean install of the app
* Starting the editor quickly in offline mode after a clean install of the app

## Regression Notes
1. Potential unintended areas of impact
N/A

2. What I did to test those areas of impact (or what existing automated tests I relied on)
N/A

3. What automated tests I added (or what prevented me from doing so)
The complexity of `EditPostActivity` makes it hard to add an automated tests for this part of the code.

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WordPress/src/main/java/org/wordpress/android/ui/posts/EditPostActivity.java [hunk 1]
```diff
@@ -2324,6 +2324,20 @@ private GutenbergPropsBuilder getGutenbergPropsBuilder() {
...
+     * Note that if the editor theme has not been initialized (usually on the first app run)
+     * the value returned is null and the `unstable_gallery_with_image_blocks` analytics property will not be reported.
+     * @return true if the the supports the new gallery block with image blocks or null if the theme is not initialized.
+     */
+    private Boolean themeSupportsGalleryWithImageBlocks() {
+        EditorTheme editorTheme = mEditorThemeStore.getEditorThemeForSite(mSite);
+        if (editorTheme == null) {
+            return null;
+        }
+        return editorTheme.getThemeSupport().getGalleryWithImageBlocks();
```
Comment (AliSoftware) [inline WordPress/src/main/java/org/wordpress/android/ui/posts/EditPostActivity.java:2338]: Question from a newbie to Kotlin, just getting curious: I was wondering if there were a way in Kotlin to do "safe navigation" on nullable values (like `&.` in ruby or `?.` in Swift)… and found [this](https://kotlinlang.org/docs/null-safety.html#safe-calls).

So would this work too?
```suggestion
        EditorTheme editorTheme = mEditorThemeStore.getEditorThemeForSite(mSite);
        return editorTheme?.getThemeSupport().getGalleryWithImageBlocks();
```

Comment (antonis) [reply WordPress/src/main/java/org/wordpress/android/ui/posts/EditPostActivity.java:2338]: Thank you for looking into this @AliSoftware 🙇 
Yep, that would work perfectly in Kotlin 😄 
Unfortunately `EditPostActivity` activity is still in Java 😞

Comment (AliSoftware) [reply WordPress/src/main/java/org/wordpress/android/ui/posts/EditPostActivity.java:2338]: 🤦‍♂️ missed that, good point 😅

**Review by develric — APPROVED** [review]
Hey @antonis 👋 , this LGTM! Even if not consistently, I was able to get the null from `getEditorThemeForSite` when using a pure self-hosted site on which I've opted out from Block Editor and especially when creating a post just after app starting up. Cannot be sure that is the only edge case, but regardless the fix worked for me 👍 .

I'll let you align with @AliSoftware to make this PR through the beta update, but let me know if any action needed from my side 🙇 .

**Review by mkevins — APPROVED** [review]
> Hey @antonis  , this LGTM! Even if not consistently, I was able to get the null from `getEditorThemeForSite` when using a pure self-hosted site on which I've opted out from Block Editor and especially when creating a post just after app starting up. Cannot be sure that is the only edge case, but regardless the fix worked for me  .
> 
> I'll let you align with @AliSoftware to make this PR through the beta update, but let me know if any action needed from my side  .

Thanks for testing this and finding steps to reproduce the original issue @develric . I had tried a few things to find steps, but nothing was reproducing it, I hadn't thought to opt out of the block editor on a self-hosted site.

Code looks good, thanks for fixing Antonis! :+1:

**Comment by antonis** [comment]
@develric, @mkevins thank you both for reviewing this so quickly 🙇 
I'll proceed with merging this in `release/18.2`

**Comment by AliSoftware** [comment]
This fix is now part of the new `18.2-rc-3` beta, thanks all 🎉
